### PR TITLE
Add import/export cmdlets for Azure Sql database

### PR DESCRIPTION
**Pull request by llali**
Add import/export and get status cmdlets for Azure Sql database. Please review and merge. Thanks

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@llali__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (llali). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/Common.ps1 [hunk 1]
```diff
@@ -250,4 +250,50 @@ function Remove-ThreatDetectionTestEnvironment ($testSuffix)
...
+}
+
+<#
+.SYNOPSIS
+Gets the parameters for import/export tests
+#>
+function Get-SqlDatabaseImportExportTestEnvironmentParameters ($testSuffix)
+{
+    $databaseName = "sql-ie-cmdlet-db" + $testSuffix;
+    $password = "Testp@ssw0rd";
```
Comment (hovsepm) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/Common.ps1]: @llali you should not hardcode passwords.

**Comment by hovsepm** [comment]
- Help missing for cmdlet Get-AzureRmSqlDatabaseImportExportStatus implemented by class Microsoft.Azure.Commands.Sql.ImportExport.Cmdlet.GetAzureSqlDatabaseImportExportStatus

- Help

 missing for cmdlet New-AzureRmSqlDatabaseExport implemented by class Microsoft.Azure.Commands.Sql.ImportExport.Cmdlet.NewAzureSqlDatabaseExport

- Help

 missing for cmdlet New-AzureRmSqlDatabaseImport implemented by class Microsoft.Azure.Commands.Sql.ImportExport.Cmdlet.NewAzureSqlDatabaseImport

**Comment by markcowl** [comment]
@azuresdkci retest this please

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/GetAzureSqlDatabaseImportExportStatus.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Azure.Commands.Sql.ImportExport.Model;
+using Microsoft.Azure.Commands.Sql.ImportExport.Service;
+
+namespace Microsoft.Azure.Commands.Sql.ImportExport.Cmdlet
+{
+    /// <summary>
+    /// Defines the AzureRmSqlDatabaseImportExportStatus cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRmSqlDatabaseImportExportStatus", ConfirmImpact = ConfirmImpact.Low)]
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/GetAzureSqlDatabaseImportExportStatus.cs]: Why ConfirmImpact.Low?  Using ConfirmImpact.Medium avoids prompts.  Also, you don't want this setting without SupportsShouldProcess=true

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/ImportExportCmdletBase.cs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    {
+        /// <summary>
+        /// Gets or sets the name of the database server to use.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "SQL Database server name.")]
+        [ValidateNotNullOrEmpty]
+        public string ServerName { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/ImportExportCmdletBase.cs]: Should have an alias to Name

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/ImportExportCmdletBase.cs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+        /// <summary>
+        /// Gets or sets the type of the storage key to use.
+        /// </summary>
+        [Parameter(Mandatory = true, HelpMessage = "The type of the storage key")]
+        [ValidateNotNullOrEmpty]
+        public StorageKeyType StorageKeyType
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/ImportExportCmdletBase.cs]: Why use these values instaed of using StorageContext?  The user can create a StorageContext using the Storage cmdlets

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseExport.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+using Microsoft.Azure.Commands.Sql.Common;
+using Microsoft.Azure.Commands.Sql.ImportExport.Model;
+using Microsoft.Azure.Commands.Sql.ImportExport.Service;
+
+namespace Microsoft.Azure.Commands.Sql.ImportExport.Cmdlet
+{
+    /// <summary>
+    /// Defines the StartAzureSqlDatabaseExport cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "AzureRmSqlDatabaseExport", ConfirmImpact = ConfirmImpact.Low)]
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseExport.cs]: Same comments. on ConfirmImpact / SupportsShouldProcess

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+using System.Threading.Tasks;
+using Microsoft.Azure.Commands.Sql.Database.Model;
+using Microsoft.Azure.Commands.Sql.ImportExport.Model;
+
+namespace Microsoft.Azure.Commands.Sql.ImportExport.Cmdlet
+{
+    /// <summary>
+    /// Defines the AzureRmSqlDatabaseImport cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "AzureRmSqlDatabaseImport", ConfirmImpact = ConfirmImpact.Low)]
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs]: Same comments.

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+            HelpMessage = "SQL Database name.")]
+        [ValidateNotNullOrEmpty]
+        public string DatabaseName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the edition of the database
+        /// </summary>
+        [Parameter(Mandatory = true, HelpMessage = "The edition of the database")]
+        [ValidateNotNull]
+        public DatabaseEdition Edition
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs]: Where would this value come from?  Is there a ValidationSet that could be used, or would this come from a Get-AzureRmDatabase call?

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+        [ValidateNotNullOrEmpty]
+        public string ServiceObjectiveName
+        {
+            get; set;
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum size for the newly imported database
+        /// </summary>
+        [Parameter(Mandatory = true, HelpMessage = "The maximum size for the newly imported database")]
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs]: What units is the maximum size in (KB, MB, GB?)

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Model/AzureSqlDatabaseImportExportBaseModel.cs [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+        /// </summary>
+        public string DatabaseName
+        {
+            get; set;
+        }
+
+        /// <summary>
+        /// Gets or sets the type of the storage key to use.
+        /// </summary>
+        public StorageKeyType StorageKeyType
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Model/AzureSqlDatabaseImportExportBaseModel.cs]: Same comment - why not use the existing StorageContext type?

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Model/StorageKeyType.cs [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+
+namespace Microsoft.Azure.Commands.Sql.ImportExport.Model
+{
+    /// <summary>
+    /// The storage key types
+    /// </summary>
+    public enum StorageKeyType
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Model/StorageKeyType.cs]: If you use StorageContext, this should not be required.

**Comment by markcowl** [comment]
@llali a few comments on the cmdlet signatures. These are suggestions, but are not yet required

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/GetAzureSqlDatabaseImportExportStatus.cs (hunk 2 above)
Comment (llali) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/GetAzureSqlDatabaseImportExportStatus.cs]: Fixed

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/ImportExportCmdletBase.cs (hunk 3 above)
Comment (llali) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/ImportExportCmdletBase.cs]: The property name form Get-AzureRmSqlServer is ServerName

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseExport.cs (hunk 5 above)
Comment (llali) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseExport.cs]: Fixed

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs (hunk 6 above)
Comment (llali) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs]: Fixed

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs (hunk 7 above)
Comment (llali) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs]: DatabaseEdition is an enum which has all the available values

File: src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs (hunk 8 above)
Comment (llali) [inline src/ResourceManager/Sql/Commands.Sql/ImportExport/Cmdlet/NewAzureSqlDatabaseImport.cs]: I added bytes to the help message

**Comment by markcowl** [comment]
@azuresdkci retest this please

**Pull request merged**